src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1261,63 +1268,84 @@ private AuthenticationResult localFlow(CallbackHandler callbackHandle,
             return cachedItem;
         }
 
+        // Trying to find refresh token first, if existed, will try to use the refresh token. 
         Logger.v(TAG, "Checking refresh tokens");
-        RefreshItem refreshItem = getRefreshToken(request);
+        RefreshItem refreshItem = getRefreshItem(request);
+        AuthenticationResult authResult = null;
```

Comment:
getRefreshToken is more accurate for this function, because what it really searches for is refreshToken and not item, even if it returns refreshtoken within item